service-base/src/main/java/org/eclipse/hono/service/registration/RegistrationService.java

```diff
@@ -41,11 +45,12 @@
      *             <li><em>404 Not Found</em> if no device with the given identifier is
      *             registered for the tenant or its <em>enabled</em> property is {@code false}.</li>
      *             </ul>
-     * @throws NullPointerException if any of the parameters is {@code null}.
+     * @throws NullPointerException if any of the parameters (except spanContext) is {@code null}.
      * @see <a href="https://www.eclipse.org/hono/api/device-registration-api/#assert-device-registration">
      *      Device Registration API - Assert Device Registration</a>
      */
-    void assertRegistration(String tenantId, String deviceId, Handler<AsyncResult<RegistrationResult>> resultHandler);
+    void assertRegistration(String tenantId, String deviceId, SpanContext spanContext,
```

Comment:
this will break the API
IMHO we should overload the method with a variant accepting a `SpanContext` instead of replacing it